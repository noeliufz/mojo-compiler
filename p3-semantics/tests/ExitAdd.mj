proc exit(i: int);

type Integer = ref int;

proc add(x, y: Integer): Integer {
  var z = New(Integer);
  z^ = x^ + y^;
  return z;
}

{
  var x = New(Integer);
  var y = New(Integer);
  var z: Integer;
  x^ = 1;
  y^ = 2;
  z = add(x,y);
  exit(z^);
}
