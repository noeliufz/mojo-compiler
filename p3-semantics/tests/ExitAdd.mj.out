BEGIN _BUILTIN
_BUILTIN.int=int
_BUILTIN.bool={false:bool=0, true:bool=1}
_BUILTIN.false:bool=0
_BUILTIN.true:bool=1
_BUILTIN.byte=[-128..127]
_BUILTIN.char={}
_BUILTIN.Null=ref null
_BUILTIN.null:Null=0
_BUILTIN.Refany=ref null
_BUILTIN.Address=ref null
_BUILTIN.Root=null object {}
_BUILTIN.Text=ref null
_BUILTIN.First:(...)
_BUILTIN.Last:(...)
_BUILTIN.Ord:(...)
_BUILTIN.Val:(...)
_BUILTIN.New:(...)
_BUILTIN.Number:(...)
_BUILTIN.Loophole:(...)
BEGIN ExitAdd
ExitAdd.exit:(VALUE i:int)
ExitAdd.Integer=ref int
ExitAdd.add:(VALUE x:Integer; VALUE y:Integer):Integer
BEGIN ExitAdd.add
ExitAdd.add.x:Integer
ExitAdd.add.y:Integer
BEGIN ExitAdd.add.1
ExitAdd.add.1.z:Integer
END ExitAdd.add.1
END ExitAdd.add
BEGIN ExitAdd.4
ExitAdd.4.x:Integer
ExitAdd.4.y:Integer
ExitAdd.4.z:Integer
END ExitAdd.4
END ExitAdd
END _BUILTIN
